plugins/adr/src/components/AdrReader/types.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * ADR content decorator function type. Decorators are responsible for
+ * performing any necessary transformations on the ADR content before rendering.
+ * @public
+ */
+export type AdrContentDecorator = (adrInfo: {
+  baseUrl: string;
+  content: string;
+}) => string;
```

Comment:
Could perhaps be wise to keep this open for additions by making the return type `{content: string}`